secrets/store_test.go

```diff
@@ -95,6 +96,61 @@ func TestNewStore(t *testing.T) {
 	}
 }
 
+func TestNewStoreMiddleware(t *testing.T) {
+	dir, err := ioutil.TempDir("", "secret_test_")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tmpFile, err := ioutil.TempFile(dir, "secrets.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+	tmpFile.Write([]byte(specificationExample))
+
+	store, err := NewStore(context.Background(), tmpFile.Name(), log.TestWrapper(t))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var (
+		expectedMiddlewareCalls = 2
+		middlewareCall          int
+		wg                      sync.WaitGroup
+	)
+
+	wg.Add(expectedMiddlewareCalls)
+	var middleware = func(next SecretHandlerFunc) SecretHandlerFunc {
+		return func(sec *Secrets) {
+			middlewareCall++
+			wg.Done()
+			next(sec)
+		}
+	}
+	store.SecretHandler(middleware, middleware)
```

Comment:
This is an example of multi-layered middleware. This can be used to handle callbacks for different secrets, the actual implementation of one or more middleware is up to the caller.